### PR TITLE
Add inline attributes for functions used in the query system

### PR DESCRIPTION
**Pull request by Zoxc**

### CODE REVIEW

File: src/librustc_query_system/query/plumbing.rs [hunk 1]
```diff
@@ -51,6 +51,7 @@ pub struct QueryState<CTX: QueryContext, C: QueryCache> {
 }
 
 impl<CTX: QueryContext, C: QueryCache> QueryState<CTX, C> {
+    #[inline]
```
Comment (Centril) [inline src/librustc_query_system/query/plumbing.rs]: This function is generic, so does it matter?

**Review by davidtwco — APPROVED** [review]
LGTM, r=me when/if comments are resolved.

File: src/librustc_query_system/query/plumbing.rs (hunk 1 above)
Comment (Zoxc) [reply src/librustc_query_system/query/plumbing.rs:54]: Yes, `#[inline]` is an inline hint, while regular generic functions are just available for inlining.

**Pull request merged**